internal/armada/server/lease.go

```diff
@@ -176,76 +187,37 @@ func (q *AggregatedQueueServer) StreamingLeaseJobs(stream api.AggregatedQueue_St
 		return errors.WithStack(err)
 	}
 
-	// Return no jobs if we don't have enough work.
-	var res common.ComputeResources = req.Resources
-	if res.AsFloat().IsLessThan(q.schedulingConfig.MinimumResourceToSchedule) {
-		return nil
-	}
-
-	queues, err := q.queueRepository.GetAllQueues()
-	if err != nil {
-		return err
-	}
-
-	activeQueues, err := q.jobRepository.FilterActiveQueues(queue.QueuesToAPI(queues))
-	if err != nil {
-		return err
-	}
-
-	usageReports, err := q.usageRepository.GetClusterUsageReports()
-	if err != nil {
-		return err
-	}
-
 	if err = q.usageRepository.UpdateClusterLeased(&req.ClusterLeasedReport); err != nil {
 		return err
 	}
 
-	// One of the below functions expects a regular lease request,
-	// so we need to convert.
-	leaseRequest := &api.LeaseRequest{
-		ClusterId:           req.ClusterId,
-		Pool:                req.Pool,
-		Resources:           req.Resources,
-		ClusterLeasedReport: req.ClusterLeasedReport,
-		MinimumJobSize:      req.MinimumJobSize,
-		Nodes:               req.Nodes,
-	}
-	nodeResources := scheduling.AggregateNodeTypeAllocations(req.Nodes)
-	clusterSchedulingInfo := scheduling.CreateClusterSchedulingInfoReport(leaseRequest, nodeResources)
-	err = q.schedulingInfoRepository.UpdateClusterSchedulingInfo(clusterSchedulingInfo)
-	if err != nil {
-		return err
+	// Return no jobs if we don't have enough work.
+	var res common.ComputeResources = req.Resources
+	if res.AsFloat().IsLessThan(q.schedulingConfig.MinimumResourceToSchedule) {
+		return nil
 	}
 
-	activeClusterReports := scheduling.FilterActiveClusters(usageReports)
-	activePoolClusterReports := scheduling.FilterPoolClusters(req.Pool, activeClusterReports)
-	activePoolCLusterIds := scheduling.GetClusterReportIds(activePoolClusterReports)
-	clusterPriorities, err := q.usageRepository.GetClusterPriorities(activePoolCLusterIds)
-	if err != nil {
-		return err
-	}
+	// Get jobs to be leased.
+	// TODO: The lease call resource reporting logic should be moved here.
+	var jobs []*api.Job
+	if rand.Float64() < q.schedulingConfig.ProbabilityOfUsingNewScheduler {
```

Comment:
we should log clearly what scheduler we've chosen

Reply:
Also if `q.schedulingConfig.ProbabilityOfUsingNewScheduler` is very low then we won't be saving usage reports often enough to stop them expiring.  I think the logic should be:

* Update usage reports (both new and old schedulers)
* Check if there's enough free resource to even bother scheduling
* If enough resource, choose a scheduler and scheudle

That way we get accurate usage reports regardless of what's going on

Reply:
Addressed